src/data_warehouse/services/duckdb_server.py

```
@@ -0,0 +1,61 @@
+
+        # Convert result to list of dictionaries
+        rows: Sequence[Sequence[Any]] = result.fetchall()
+        columns = result.description
+        if columns:
+            column_names = [col[0] for col in columns]
+            return [dict(zip(column_names, row, strict=False)) for row in rows]
+        return []
+
+    except Exception as e:
+        logger.error(f"Query execution error: {str(e)}")
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)